## Pobranie danych API 'https://dane.gov.pl/pl'


In [2]:
import json
import requests
import pandas as pd

In [4]:
url = 'https://public-esa.ose.gov.pl/api/v1/smog'
response = requests.get(url)
print(response.status_code)

200


In [13]:
data = response.json()
print(type(data))
print(data.keys())

<class 'dict'>
dict_keys(['smog_data', 'it_has_next_page', 'pages_total'])


In [15]:
df = pd.json_normalize(data['smog_data'])
df.head()

timestamp                                        school.name  \
0  2025-05-15 17:42:41                  SZKOŁA PODSTAWOWA WE WRZĄSOWICACH   
1  2025-05-15 17:42:41  PUBLICZNA SZKOŁA PODSTAWOWA NR 2 IM. KAZIMIERZ...   
2  2025-05-15 17:42:41                      ZESPÓŁ SZKÓŁ NR 1 W PSZCZYNIE   
3  2025-05-15 17:42:41  SPECJALNY OŚRODEK WYCHOWAWCZY NR 1 W OSTROWCU ...   
4  2025-05-15 17:42:41  SZKOŁA PODSTAWOWA IM. ORŁA BIAŁEGO W KUROWICAC...   

              school.street school.post_code              school.city  \
0               UL. SZKOLNA           32-040               WRZĄSOWICE   
1   UL. WAWRZYŃCA ŚWIERZEGO           47-100        STRZELCE OPOLSKIE   
2  UL. KAZIMIERZA WIELKIEGO           43-200                 PSZCZYNA   
3                   SZKOLNA           27-400  OSTROWIEC ŚWIĘTOKRZYSKI   
4                   SZKOLNA           95-006              WOLA RAKOWA   

  school.longitude school.latitude  data.humidity_avg  data.pressure_avg  \
0         19.94282        49.96103          51.180000         971.600000   
1        18.314889       50.503431          61.666667        1026.000000   
2       18.9457058       49.965883          57.000000         988.600000   
3        21.418144         50.9312          91.970000         995.090000   
4         19.61157        51.67447          60.700000         986.133333   

   data.temperature_avg  data.pm10_avg  data.pm25_avg  
0             11.140000       2.980000       2.680000  
1              8.733333       1.000000       0.573333  
2             12.566667       2.466667       1.506667  
3              5.000000      44.020000      25.100000  
4             11.800000      33.833333      17.713333

In [16]:
df.columns

Index(['timestamp', 'school.name', 'school.street', 'school.post_code',
       'school.city', 'school.longitude', 'school.latitude',
       'data.humidity_avg', 'data.pressure_avg', 'data.temperature_avg',
       'data.pm10_avg', 'data.pm25_avg'],
      dtype='object')

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1184 entries, 0 to 1183
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   timestamp             1184 non-null   object 
 1   school.name           1184 non-null   object 
 2   school.street         945 non-null    object 
 3   school.post_code      1184 non-null   object 
 4   school.city           1184 non-null   object 
 5   school.longitude      1184 non-null   object 
 6   school.latitude       1184 non-null   object 
 7   data.humidity_avg     1182 non-null   float64
 8   data.pressure_avg     1182 non-null   float64
 9   data.temperature_avg  1182 non-null   float64
 10  data.pm10_avg         1184 non-null   float64
 11  data.pm25_avg         1184 non-null   float64
dtypes: float64(5), object(7)
memory usage: 111.1+ KB


In [18]:
df.isnull().sum()

timestamp                 0
school.name               0
school.street           239
school.post_code          0
school.city               0
school.longitude          0
school.latitude           0
data.humidity_avg         2
data.pressure_avg         2
data.temperature_avg      2
data.pm10_avg             0
data.pm25_avg             0
dtype: int64

## Wnioski

Dane przedstawione w analizie posiadają braki w kolumnach: school.street (239), data.humidity_avg(2), data.pressure_avg(2), data.temperature_avg (2). Dane dotyczą adresów szkoły z dokładnością do szerokości geograficznej.

## Co udało się zrobić?:
- zaimportowalem dane json
- rozbiłem zagnieżdżenie
- zweryfikowałem jakość danych